packages/authgear-react-native/ios/AGAuthgearReactNative.m

```diff
@@ -254,22 +253,51 @@ + (BOOL)application:(UIApplication *)application
-    wv.navigationDelegate = self;
-    [wv loadRequest:[NSURLRequest requestWithURL:url]];
-    [vc.view addSubview:wv];
-    vc.navigationItem.rightBarButtonItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemDone target:self action:@selector(dismissWebView)];
-    self.webViewViewController = vc;
-
-    UINavigationController *nav = [[UINavigationController alloc] initWithRootViewController:vc];
-    nav.modalPresentationStyle = UIModalPresentationPageSheet;
-
-    UIViewController *rootViewController = RCTPresentedViewController();
-    [rootViewController presentViewController:nav animated:YES completion:nil];
-    resolve(nil);
+    if (@available(iOS 12.0, *)) {
+        self.asSession = [[ASWebAuthenticationSession alloc] initWithURL:url
+                                                                            callbackURLScheme:scheme
+                                                                            completionHandler:^(NSURL *url, NSError *error) {
+            [AGAuthgearReactNative unregisterCurrentWechatRedirectURI];
+            if (error) {
+                BOOL isUserCancelled = ([[error domain] isEqualToString:ASWebAuthenticationSessionErrorDomain] &&
+                [error code] == ASWebAuthenticationSessionErrorCodeCanceledLogin);
+                if (isUserCancelled) {
+                    reject(@"CANCEL", @"CANCEL", error);
+                } else {
+                    reject(RCTErrorUnspecified, [NSString stringWithFormat:@"Unable to open URL: %@", url], error);
+                }
+            } else {
+                resolve(nil);
+            }
+            [self cleanup];
+        }];
+        if (@available(iOS 13.0, *)) {
+            self.asSession.presentationContextProvider = self;
+            self.asSession.prefersEphemeralWebBrowserSession = @YES;
+        }
+        [self.asSession start];
+    } else if (@available(iOS 11.0, *)) {
+        self.sfSession = [[SFAuthenticationSession alloc] initWithURL:url
+                                                                      callbackURLScheme:scheme
+                                                                      completionHandler:^(NSURL *url, NSError *error) {
+            [AGAuthgearReactNative unregisterCurrentWechatRedirectURI];
+            if (error) {
+                BOOL isUserCancelled = ([[error domain] isEqualToString:SFAuthenticationErrorDomain] &&
+                [error code] == SFAuthenticationErrorCanceledLogin);
+                if (isUserCancelled) {
+                    reject(@"CANCEL", @"CANCEL", error);
+                } else {
+                    reject(RCTErrorUnspecified, [NSString stringWithFormat:@"Unable to open URL: %@", url], error);
+                }
+            } else {
+                resolve(nil);
+            }
+            [self cleanup];
+        }];
+        [self.sfSession start];
+    }
```

Comment:
Same question as the iOS version, ref https://github.com/authgear/authgear-sdk-ios/pull/78#discussion_r700674785